src/librustc/hir/mod.rs

```diff
@@ -229,21 +228,34 @@ pub enum LifetimeName {
     Static,
 
     /// Some user-given name like `'x`
-    Name(Name),
+    Ident(Ident),
 }
 
 impl LifetimeName {
-    pub fn name(&self) -> Name {
-        use self::LifetimeName::*;
+    pub fn ident(&self) -> Ident {
+        match *self {
+            LifetimeName::Implicit => keywords::Invalid.ident(),
+            LifetimeName::Fresh(_) | LifetimeName::Underscore =>
+                keywords::UnderscoreLifetime.ident(),
+            LifetimeName::Static => keywords::StaticLifetime.ident(),
+            LifetimeName::Ident(ident) => ident,
+        }
+    }
+
+    pub fn modern(&self) -> LifetimeName {
```

Comment:
I just checked and "modern" doesn't seem to be documented anywhere. Might be a good idea to point everything to `Span` (or even `SyntaxContext`)'s `modern` method, and have that documented.

Reply:
OK, will document.
The name itself is terribly generic, but it does a relatively intuitive thing - normalizes an identifier "up to macros 2.0", i.e. identifiers coming from different `macro_rules!` macros or from the non-macro context become same, but identifiers coming from different `macro` macros still stay different.

Reply:
Hmm, so this normalizes for "global" or "item" hygiene, as opposed to "local" hygiene?

Reply:
Yes.

Reply:
I have some ideas how to make this less error-prone (so that `.modern()` is never forgotten accidentally) by removing `PartialEq` impl for `Ident` and introducing a couple of comparable newtypes, but this requires some prerequisite work.